src/tablestore.rs

```diff
@@ -96,7 +108,7 @@ impl TableStore {
             filter_cache: RwLock::new(HashMap::new()),
             transactional_wal_store: Arc::new(DelegatingTransactionalObjectStore::new(
```

Comment:
May be I am missing something.
`transactional_wal_store` is used for both compacted ssts and wal ssts. It is not using `CacheableObjectStore`, how do the files gets written to disk?

Reply:
they're still using the original `ObjectStore` interface, we can switch it to CacheableObjectStore if we want to cache these files.

if i understand correctly in https://github.com/slatedb/slatedb/pull/139/files#r1740202335 , we do not need to cache the WAL files as they're already cached in memory, right?

Reply:
We do not need to cache WAL. This is used for more than WAL though. `write_sst` writes L0 as well, and it uses `transactional_wal_store`. Naming should be changed.